Restart adapters prior update on windows

Hallo,

bei einen Updates von bestimmten Adapter kommt immer der Code 25.
Wird der jeweilige Adapter **vorher** gestoppt funktioniert das Update bzw. die Installation.

Sollte generell bei Windows Systemen die Adapter vorher gestoppt werden?

LG
Sigi234

Plattform: Windows
RAM: 16 GB 
Node.js: 12.22.5
NPM: 6.14.14
JS Controller: 3.3.15
Admin: 5.1.25
Web: 3.4.9
Socket.io: 3.1.4
Script Engine: 5.2.8

> Log wäre interessant.

`


2021-08-23 20:24:14.747 - info: host.Medion(Home) iobroker url https://github.com/iobroker-community-adapters/iobroker.kodi.git --host Medion(Home)
--
2021-08-23 20:24:16.675 - info: host.Medion(Home) iobroker install iobroker-community-adapters/iobroker.kodi#8b0e99796771e99d5893e111ba5956d8716efaea
2021-08-23 20:24:17.420 - info: host.Medion(Home) iobroker NPM version: 6.14.14
2021-08-23 20:24:17.421 - info: host.Medion(Home) iobroker npm install iobroker-community-adapters/iobroker.kodi#8b0e99796771e99d5893e111ba5956d8716efaea --loglevel error (System call)
2021-08-23 20:24:44.765 - info: host.Medion(Home) iobroker npm ERR! code EBUSY
npm ERR! syscall rename
npm ERR! path C:\Program Files\iobroker\Home\node_modules\iobroker.kodi
npm ERR! dest C:\Program Files\iobroker\Home\node_modules\.iobroker.kodi.DELETE
npm ERR! errno -4082
2021-08-23 20:24:44.767 - info: host.Medion(Home) iobroker npm ERR! EBUSY: resource busy or locked, rename 'C:\Program Files\iobroker\Home\node_modules\iobroker.kodi' -> 'C:\Program Files\iobroker\Home\node_modules\.iobroker.kodi.DELETE'
2021-08-23 20:24:44.779 - info: host.Medion(Home) iobroker
2021-08-23 20:24:44.780 - info: host.Medion(Home) iobroker npm ERR! A complete log of this run can be found in:
npm ERR! C:\Program Files\iobroker\Home\env\npm-cache\_logs\2021-08-23T18_24_44_773Z-debug.log
2021-08-23 20:24:44.810 - error: host.Medion(Home) iobroker host.Medion(Home) Cannot install iobroker-community-adapters/iobroker.kodi#8b0e99796771e99d5893e111ba5956d8716efaea: 4294963214
2021-08-23 20:24:45.827 - info: host.Medion(Home) iobroker exit 25

`

--- a/packages/controller/lib/setup/setupInstall.js
+++ b/packages/controller/lib/setup/setupInstall.js
@@ -136,8 +136,11 @@ function Install(options) {
         const sources = repoUrl;
         options.unsafePerm = sources[packetName] && sources[packetName].unsafePerm;
 
-        // Check if flag stopBeforeUpdate is true
-        if (sources[packetName] && sources[packetName].stopBeforeUpdate && !stoppedList) {
+        // Check if flag stopBeforeUpdate is true or on windows we stop because of issue #1436
+        if (
+            ((sources[packetName] && sources[packetName].stopBeforeUpdate) || process.platform === 'win32') &&
+            !stoppedList
+        ) {
             const arr = await objects.getObjectListAsync({
                 startkey: `system.adapter.${packetName}.`,
                 endkey: `system.adapter.${packetName}.\u9999`
@@ -146,10 +149,10 @@ function Install(options) {
             stoppedList = [];
 
             if (arr) {
-                for (let id = 0; id < arr.rows.length; id++) {
+                for (const row of arr.rows) {
                     // stop only started instances on this host
-                    if (arr.rows[id].value.common.enabled && hostname === arr.rows[id].value.common.host) {
-                        stoppedList.push(arr.rows[id].value);
+                    if (row.value.common.enabled && hostname === row.value.common.host) {
+                        stoppedList.push(row.value);
                     }
                 }
             }

